File: Solutions/Kamikoto66/GameShop3/Potion.cpp
#include "Item.h"
#include "Potion.h"
#include <iostream>


Potion::Potion()
{
}

Potion::Potion(string name, string description, int weight, int value, string type, int capacity)
	:Item(name, description, weight, value), _type(type), _capacity(capacity)
{
}


Potion::~Potion()
{
}

void Potion::Describe()
{
	Item::Describe();
	cout << "Type = " << _type.c_str() << endl;
	cout << "Capacity = " << _capacity << endl;
	cout << endl;
}

File: Solutions/whiteWeed/Game Shop/4/Potion.java
/**
 * 
 */
package whiteWeed.gameShop;

/**
 * <pre>
 * whiteWeed.gameShop
 *   |_Potion
 * 
 * 1. 개요 : 
 * 2. 작성일 : 2017. 6. 5.
 * </pre>
 * 
 * @author				:  Weed
 * @version				:  1.0
 */
public class Potion extends Item {
	String type;
	int capacity;
	
	public Potion(String name, String description, int weight, int value, String type, int capacity){
		super(name, description, weight, value);
		this.type = type;
		this.capacity = capacity;
	}
	
	public void describe(){
		super.describe();
		System.out.println("Type          = " + this.type);
		System.out.println("Capacity      = " + this.capacity);
	}
}
